src/VerifyU2FController.js

```diff
@@ -0,0 +1,109 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/*global u2f */
+
+define([
+  'okta',
+  'util/FormController',
+  'util/FormType',
+  'views/shared/FooterSignout',
+  'vendor/lib/q',
+  'u2f-api-polyfill'
+],
+function (Okta, FormController, FormType, FooterSignout, Q) {
+
+  var _ = Okta._;
+
+  return FormController.extend({
+    className: 'verify-u2f',
+    Model: {
+      save: function () {
+        this.trigger('request');
+
+        return this.doTransaction(function (transaction) {
+          var factor = _.findWhere(transaction.factors, {
+            factorType: 'u2f',
+            provider: 'FIDO'
+          });
+          var self = this;
+          return factor.verify()
+          .then(function (transaction) {
+            var factorData = transaction.factor;
+            var appId = factorData.profile.appId;
+            var registeredKeys = [{version: factorData.profile.version, keyHandle: factorData.profile.credentialId }];
+            self.trigger('request');
+
+            var deferred = Q.defer();
+            u2f.sign(appId, factorData.challenge.nonce, registeredKeys, function (data) {
+              self.trigger('errors:clear');
+              if (data.errorCode && data.errorCode !== 0) {
+                deferred.reject({ responseJSON: {errorSummary: 'Error Code: ' + data.errorCode}});
+              } else {
+                return factor.verify({
+                  clientData: data.clientData,
+                  signatureData: data.signatureData
+                })
+                .then(deferred.resolve);
+              }
+            });
+            return deferred.promise;
+          });
+        });
+      }
+    },
+
+    Form: {
+      autoSave: true,
+      hasSavingState: false,
+      title: Okta.loc('factor.u2f', 'login'),
+      className: 'verify-u2f-form',
+      noCancelButton: true,
+      save: Okta.loc('verify.u2f.retry', 'login'),
+      modelEvents: function () {
+        return {
+          'error': '_showRetry',
+          'request': '_hideRetry'
+        };
+      },
+
+      formChildren: function () {
+        var result = [];
+        result.push(
+          FormType.View({
+            View: '<p>' + Okta.loc('verify.u2f.instructions', 'login') + '</p>\
+                  <p>' + Okta.loc('verify.u2f.instructionsBluetooth', 'login') + '</p>\
+                  <div data-se="u2f-waiting" class="okta-waiting-spinner"></div>'
+          })
+        );
+        return result;
+      },
+
+      postRender: function () {
+        this.model.save();
+      },
+
+      _showRetry: function () {
+        this.$('.okta-waiting-spinner').hide();
```

Comment:
Can we control this through a class intead? I.e. add the class here and remove it in the other one, and then have the class deal with what is showing.